### PR TITLE
Cloud examples using druid-k8s-extension

### PR DESCRIPTION
**Pull request by AdheipSingh**
### Description

- cloud example for running druid with deepstorage using digital ocean spaces ( object store )
- using druid-k8s-extensions 

will add more cloud examples in this dir.

<hr>

This PR has:
- [x] been tested on a real K8S cluster to ensure creation of a brand new Druid cluster works.
- [x] been tested for backward compatibility on a real K*S cluster by applying the changes introduced here on an existing Druid cluster. If there are any backward incompatible changes then they have been noted in the PR description.
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [ ] added documentation for new or modified features or behaviors.

<hr>

##### Key changed/added files in this PR
 * `examples`

cc @himanshug @nishantmonu51

### CODE REVIEW

**Comment by AdheipSingh** [comment]
@himanshug pls review these configs. :)

File: examples/cloud/do-druid.yaml [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+      druid.port: 8080
+      nodeConfigMountPath: "/opt/druid/conf/druid/cluster/query/broker"
+      replicas: 1
+      runtime.properties: |
+        druid.service=druid/broker
+        # HTTP server threads
+        druid.broker.http.numConnections=1
+        druid.server.http.numThreads=1
+        # Processing threads and buffers
+        druid.processing.buffer.sizeBytes=1
```
Comment (nishantmonu51) [inline examples/cloud/do-druid.yaml:108]: 128M or 64M, 1 byte is unusable :)

File: examples/cloud/do-druid.yaml [hunk 2]
```diff
@@ -0,0 +1,224 @@
+apiVersion: "druid.apache.org/v1alpha1"
+kind: "Druid"
+metadata:
+  name: do-cluster
+spec:
+  image: apache/druid:0.21.1
+  startScript: /druid.sh
+  podLabels:
```
Comment (AdheipSingh) [inline examples/cloud/do-druid.yaml:8]: update druid image

**Comment by renatocron** [comment]
I've tried running this on 22.1 but now logs are full of NullPointer exceptions

[druid-druid-gl-coordinators-0] 2022-04-04T15:48:00,229 WARN [HttpClient-Netty-Boss-0] org.jboss.netty.channel.SimpleChannelUpstreamHandler - EXCEPTION, please implement org.jboss.netty.handler.codec.http.HttpContentDecompressor.exceptionCaught() for proper handling. 
[druid-druid-gl-coordinators-0] org.jboss.netty.channel.ConnectTimeoutException: connection timed out: /10.244.1.13:8088 
[druid-druid-gl-coordinators-0] 	at org.jboss.netty.channel.socket.nio.NioClientBoss.processConnectTimeout(NioClientBoss.java:139) [netty-3.10.6.Final.jar:?] 
[druid-druid-gl-coordinators-0] 	at org.jboss.netty.channel.socket.nio.NioClientBoss.process(NioClientBoss.java:83) [netty-3.10.6.Final.jar:?] 
[druid-druid-gl-coordinators-0] 	at org.jboss.netty.channel.socket.nio.AbstractNioSelector.run(AbstractNioSelector.java:337) [netty-3.10.6.Final.jar:?] 
[druid-druid-gl-coordinators-0] 	at org.jboss.netty.channel.socket.nio.NioClientBoss.run(NioClientBoss.java:42) [netty-3.10.6.Final.jar:?] 
[druid-druid-gl-coordinators-0] 	at org.jboss.netty.util.ThreadRenamingRunnable.run(ThreadRenamingRunnable.java:108) [netty-3.10.6.Final.jar:?] 
[druid-druid-gl-coordinators-0] 	at org.jboss.netty.util.internal.DeadLockProofWorker$1.run(DeadLockProofWorker.java:42) [netty-3.10.6.Final.jar:?] 
[druid-druid-gl-coordinators-0] 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_275] 
[druid-druid-gl-coordinators-0] 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_275] 
[druid-druid-gl-coordinators-0] 	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_275] 
[druid-druid-gl-coordinators-0] 2022-04-04T15:48:00,227 ERROR [HttpServerInventoryView-3] org.apache.druid.server.coordination.ChangeRequestHttpSyncer - failed to get sync response from [http://10.244.1.13:8088/_1648830676700]. Return code [0], Reason: [null] 
[druid-druid-gl-coordinators-0] org.jboss.netty.channel.ChannelException: Faulty channel in resource pool 
[druid-druid-gl-coordinators-0] 	at org.apache.druid.java.util.http.client.NettyHttpClient.go(NettyHttpClient.java:131) ~[druid-core-0.22.1.jar:0.22.1] 
[druid-druid-gl-coordinators-0] 	at org.apache.druid.server.coordination.ChangeRequestHttpSyncer.sync(ChangeRequestHttpSyncer.java:205) ~[druid-server-0.22.1.jar:0.22.1] 
[druid-druid-gl-coordinators-0] 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_275] 
[druid-druid-gl-coordinators-0] 	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_275] 
[druid-druid-gl-coordinators-0] 	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) [?:1.8.0_275] 
[druid-druid-gl-coordinators-0] 	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) [?:1.8.0_275] 
[druid-druid-gl-coordinators-0] 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_275] 
[druid-druid-gl-coordinators-0] 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_275] 
[druid-druid-gl-coordinators-0] 	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_275] 
[druid-druid-gl-coordinators-0] Caused by: org.jboss.netty.channel.ConnectTimeoutException: connection timed out: /10.244.1.13:8088 
[druid-druid-gl-coordinators-0] 	at org.jboss.netty.channel.socket.nio.NioClientBoss.processConnectTimeout(NioClientBoss.java:139) ~[netty-3.10.6.Final.jar:?] 
[druid-druid-gl-coordinators-0] 	at org.jboss.netty.channel.socket.nio.NioClientBoss.process(NioClientBoss.java:83) ~[netty-3.10.6.Final.jar:?] 
[druid-druid-gl-coordinators-0] 	at org.jboss.netty.channel.socket.nio.AbstractNioSelector.run(AbstractNioSelector.java:337) ~[netty-3.10.6.Final.jar:?] 
[druid-druid-gl-coordinators-0] 	at org.jboss.netty.channel.socket.nio.NioClientBoss.run(NioClientBoss.java:42) ~[netty-3.10.6.Final.jar:?] 
[druid-druid-gl-coordinators-0] 	at org.jboss.netty.util.ThreadRenamingRunnable.run(ThreadRenamingRunnable.java:108) ~[netty-3.10.6.Final.jar:?] 
[druid-druid-gl-coordinators-0] 	at org.jboss.netty.util.internal.DeadLockProofWorker$1.run(DeadLockProofWorker.java:42) ~[netty-3.10.6.Final.jar:?] 
[druid-druid-gl-coordinators-0] 	... 3 more 

[druid-druid-gl-routers-1] 2022-04-04T15:48:01,279 ERROR [org.apache.druid.k8s.discovery.K8sDruidNodeDiscoveryProvider$NodeRoleWatcherbroker] org.apache.druid.k8s.discovery.K8sDruidNodeDiscoveryProvider$NodeRoleWatcher - Error while watching node type [BROKER] 
[druid-druid-gl-routers-1] java.lang.NullPointerException: null

**Comment by jphavoc** [comment]
> I've tried running this on 22.1 but now logs are full of NullPointer exceptions
> 
> [druid-druid-gl-coordinators-0] 2022-04-04T15:48:00,229 WARN [HttpClient-Netty-Boss-0] org.jboss.netty.channel.SimpleChannelUpstreamHandler - EXCEPTION, please implement org.jboss.netty.handler.codec.http.HttpContentDecompressor.exceptionCaught() for proper handling. [druid-druid-gl-coordinators-0] org.jboss.netty.channel.ConnectTimeoutException: connection timed out: /10.244.1.13:8088 [druid-druid-gl-coordinators-0] at org.jboss.netty.channel.socket.nio.NioClientBoss.processConnectTimeout(NioClientBoss.java:139) [netty-3.10.6.Final.jar:?] [druid-druid-gl-coordinators-0] at org.jboss.netty.channel.socket.nio.NioClientBoss.process(NioClientBoss.java:83) [netty-3.10.6.Final.jar:?] [druid-druid-gl-coordinators-0] at org.jboss.netty.channel.socket.nio.AbstractNioSelector.run(AbstractNioSelector.java:337) [netty-3.10.6.Final.jar:?] [druid-druid-gl-coordinators-0] at org.jboss.netty.channel.socket.nio.NioClientBoss.run(NioClientBoss.java:42) [netty-3.10.6.Final.jar:?] [druid-druid-gl-coordinators-0] at org.jboss.netty.util.ThreadRenamingRunnable.run(ThreadRenamingRunnable.java:108) [netty-3.10.6.Final.jar:?] [druid-druid-gl-coordinators-0] at org.jboss.netty.util.internal.DeadLockProofWorker$1.run(DeadLockProofWorker.java:42) [netty-3.10.6.Final.jar:?] [druid-druid-gl-coordinators-0] at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_275] [druid-druid-gl-coordinators-0] at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_275] [druid-druid-gl-coordinators-0] at java.lang.Thread.run(Thread.java:748) [?:1.8.0_275] [druid-druid-gl-coordinators-0] 2022-04-04T15:48:00,227 ERROR [HttpServerInventoryView-3] org.apache.druid.server.coordination.ChangeRequestHttpSyncer - failed to get sync response from [http://10.244.1.13:8088/_1648830676700]. Return code [0], Reason: [null] [druid-druid-gl-coordinators-0] org.jboss.netty.channel.ChannelException: Faulty channel in resource pool [druid-druid-gl-coordinators-0] at org.apache.druid.java.util.http.client.NettyHttpClient.go(NettyHttpClient.java:131) ~[druid-core-0.22.1.jar:0.22.1] [druid-druid-gl-coordinators-0] at org.apache.druid.server.coordination.ChangeRequestHttpSyncer.sync(ChangeRequestHttpSyncer.java:205) ~[druid-server-0.22.1.jar:0.22.1] [druid-druid-gl-coordinators-0] at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_275] [druid-druid-gl-coordinators-0] at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_275] [druid-druid-gl-coordinators-0] at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) [?:1.8.0_275] [druid-druid-gl-coordinators-0] at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) [?:1.8.0_275] [druid-druid-gl-coordinators-0] at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_275] [druid-druid-gl-coordinators-0] at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_275] [druid-druid-gl-coordinators-0] at java.lang.Thread.run(Thread.java:748) [?:1.8.0_275] [druid-druid-gl-coordinators-0] Caused by: org.jboss.netty.channel.ConnectTimeoutException: connection timed out: /10.244.1.13:8088 [druid-druid-gl-coordinators-0] at org.jboss.netty.channel.socket.nio.NioClientBoss.processConnectTimeout(NioClientBoss.java:139) ~[netty-3.10.6.Final.jar:?] [druid-druid-gl-coordinators-0] at org.jboss.netty.channel.socket.nio.NioClientBoss.process(NioClientBoss.java:83) ~[netty-3.10.6.Final.jar:?] [druid-druid-gl-coordinators-0] at org.jboss.netty.channel.socket.nio.AbstractNioSelector.run(AbstractNioSelector.java:337) ~[netty-3.10.6.Final.jar:?] [druid-druid-gl-coordinators-0] at org.jboss.netty.channel.socket.nio.NioClientBoss.run(NioClientBoss.java:42) ~[netty-3.10.6.Final.jar:?] [druid-druid-gl-coordinators-0] at org.jboss.netty.util.ThreadRenamingRunnable.run(ThreadRenamingRunnable.java:108) ~[netty-3.10.6.Final.jar:?] [druid-druid-gl-coordinators-0] at org.jboss.netty.util.internal.DeadLockProofWorker$1.run(DeadLockProofWorker.java:42) ~[netty-3.10.6.Final.jar:?] [druid-druid-gl-coordinators-0] ... 3 more
> 
> [druid-druid-gl-routers-1] 2022-04-04T15:48:01,279 ERROR [org.apache.druid.k8s.discovery.K8sDruidNodeDiscoveryProvider$NodeRoleWatcherbroker] org.apache.druid.k8s.discovery.K8sDruidNodeDiscoveryProvider$NodeRoleWatcher - Error while watching node type [BROKER] [druid-druid-gl-routers-1] java.lang.NullPointerException: null

I tried it with 0.23.0 and its works fine for me.

**Comment by AdheipSingh** [comment]
@renatocron not sure if this extension is being maintained or been upgraded to latest k8s API.
Will close this PR.

**Pull request closed without merging**